pkg/webhook/resourceinterpretercustomization/helper.go

```diff
@@ -0,0 +1,50 @@
+package resourceinterpretercustomization
+
+import (
+	"fmt"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+)
+
+//nolint:gocyclo
+func validateCustomizationRule(oldRules, newRules configv1alpha1.ResourceInterpreterCustomization) bool {
+	if oldRules.Spec.Target.APIVersion != newRules.Spec.Target.APIVersion ||
+		oldRules.Spec.Target.Kind != newRules.Spec.Target.Kind {
+		return true
+	}
+	if oldRules.Spec.Customizations.Retention != nil && newRules.Spec.Customizations.Retention != nil {
+		return false
+	}
+	if oldRules.Spec.Customizations.ReplicaResource != nil && newRules.Spec.Customizations.ReplicaResource != nil {
+		return false
+	}
+	if oldRules.Spec.Customizations.ReplicaRevision != nil && newRules.Spec.Customizations.ReplicaRevision != nil {
+		return false
+	}
+	if oldRules.Spec.Customizations.StatusReflection != nil && newRules.Spec.Customizations.StatusReflection != nil {
+		return false
+	}
+	if oldRules.Spec.Customizations.StatusAggregation != nil && newRules.Spec.Customizations.StatusAggregation != nil {
+		return false
+	}
+	if oldRules.Spec.Customizations.HealthInterpretation != nil && newRules.Spec.Customizations.HealthInterpretation != nil {
+		return false
+	}
+	if oldRules.Spec.Customizations.DependencyInterpretation != nil && newRules.Spec.Customizations.DependencyInterpretation != nil {
+		return false
+	}
+	return true
+}
+
+func validateResourceInterpreterCustomizations(customization *configv1alpha1.ResourceInterpreterCustomization, customizations *configv1alpha1.ResourceInterpreterCustomizationList) error {
+	for _, config := range customizations.Items {
+		//skip self verification
+		if config.Name == customization.Name {
+			continue
+		}
+		if !validateCustomizationRule(config, *customization) {
+			return fmt.Errorf("conflicting with existing ResourceInterpreterCustomization(%s)", config.Name)
```

Comment:
Can here log the conflict here?

Reply:
> conflict operation

Do you mean `req.Operation`  ?

Reply:
> Can here log the conflict operation here?

+1
I think this information is useful for users.

Reply:
For example: `retention`.